File: sdks/php/src/Resources/SuppliersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Supplier;
use Puxbay\Models\PaginatedResponse;

class SuppliersResource extends BaseResource
{
    public function list(int $page = 1, int $pageSize = 20): PaginatedResponse
    {
        $data = $this->get("suppliers/?page={$page}&page_size={$pageSize}");
        return PaginatedResponse::fromArray($data, fn($item) => Supplier::fromArray($item));
    }

    public function getSupplier(string $supplierId): Supplier
    {
        $data = $this->get("suppliers/{$supplierId}/");
        return Supplier::fromArray($data);
    }

    public function create(Supplier $supplier): Supplier
    {
        $data = $this->post('suppliers/', $supplier->toArray());
        return Supplier::fromArray($data);
    }

    public function update(string $supplierId, Supplier $supplier): Supplier
    {
        $data = $this->patch("suppliers/{$supplierId}/", $supplier->toArray());
        return Supplier::fromArray($data);
    }

    public function deleteSupplier(string $supplierId): void
    {
        $this->delete("suppliers/{$supplierId}/");
    }
}

File: sdks/dotnet/src/Resources/SuppliersResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class SuppliersResource
    {
        private readonly PuxbayClient _client;
        public SuppliersResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Supplier>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await _client.GetAsync<PaginatedResponse<Supplier>>($"suppliers/?page={page}&page_size={pageSize}");
        }

        public async Task<Supplier> GetAsync(string supplierId)
        {
            return await _client.GetAsync<Supplier>($"suppliers/{supplierId}/");
        }

        public async Task<Supplier> CreateAsync(Supplier supplier)
        {
            return await _client.PostAsync<Supplier>("suppliers/", supplier);
        }

        public async Task<Supplier> UpdateAsync(string supplierId, Supplier supplier)
        {
            return await _client.PatchAsync<Supplier>($"suppliers/{supplierId}/", supplier);
        }

        public async Task DeleteAsync(string supplierId)
        {
            await _client.DeleteAsync($"suppliers/{supplierId}/");
        }
    }
}
